File: android/src/main/java/cn/qiuxiang/react/amap3d/AMapPolylineManager.java
package cn.qiuxiang.react.amap3d;

import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewGroupManager;
import com.facebook.react.uimanager.annotations.ReactProp;

class AMapPolylineManager extends ViewGroupManager<AMapPolyline> {
    @Override
    public String getName() {
        return "AMapPolyline";
    }

    @Override
    protected AMapPolyline createViewInstance(ThemedReactContext reactContext) {
        return new AMapPolyline(reactContext);
    }

    @ReactProp(name = "coordinates")
    public void setCoordinate(AMapPolyline polyline, ReadableArray coordinates) {
        polyline.setCoordinates(coordinates);
    }

    @ReactProp(name = "colors")
    public void setColors(AMapPolyline polyline, ReadableArray colors) {
        polyline.setColors(colors);
    }

    @ReactProp(name = "color", customType = "Color")
    public void setColor(AMapPolyline polyline, int color) {
        polyline.setColor(color);
    }

    @ReactProp(name = "width")
    public void setWidth(AMapPolyline polyline, int width) {
        polyline.setWidth(width);
    }

    @ReactProp(name = "zIndex")
    public void setZIndex(AMapPolyline polyline, int zIndex) {
        polyline.setZIndex(zIndex);
    }

    @ReactProp(name = "opacity")
    public void setOpacity(AMapPolyline polyline, float opacity) {
        polyline.setOpacity(opacity);
    }

    @ReactProp(name = "geodesic")
    public void setGeodesic(AMapPolyline polyline, boolean geodesic) {
        polyline.setGeodesic(geodesic);
    }

    @ReactProp(name = "dottedLine")
    public void setDottedLine(AMapPolyline polyline, boolean dottedLine) {
        polyline.setDottedLine(dottedLine);
    }

    @ReactProp(name = "gradient")
    public void setGradient(AMapPolyline polyline, boolean gradient) {
        polyline.setGradient(gradient);
    }
}

File: android/src/main/java/cn/qiuxiang/react/amap3d/AMapPolylineManager.kt
package cn.qiuxiang.react.amap3d

import com.facebook.react.bridge.ReadableArray
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewGroupManager
import com.facebook.react.uimanager.annotations.ReactProp

@Suppress("unused")
internal class AMapPolylineManager : ViewGroupManager<AMapPolyline>() {
    override fun getName(): String {
        return "AMapPolyline"
    }

    override fun createViewInstance(reactContext: ThemedReactContext): AMapPolyline {
        return AMapPolyline(reactContext)
    }

    override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any> {
        return mapOf("onPolylineClick" to mapOf("registrationName" to "onPolylineClick"))
    }

    @ReactProp(name = "coordinates")
    fun setCoordinate(polyline: AMapPolyline, coordinates: ReadableArray) {
        polyline.setCoordinates(coordinates)
    }

    @ReactProp(name = "colors")
    fun setColors(polyline: AMapPolyline, colors: ReadableArray) {
        polyline.setColors(colors)
    }

    @ReactProp(name = "color", customType = "Color")
    fun setColor(polyline: AMapPolyline, color: Int) {
        polyline.setColor(color)
    }

    @ReactProp(name = "width")
    fun setWidth(polyline: AMapPolyline, width: Int) {
        polyline.setWidth(width.toFloat())
    }

    @ReactProp(name = "zIndex")
    fun setZIndex(polyline: AMapPolyline, zIndex: Int) {
        polyline.setZIndex(zIndex.toFloat())
    }

    @ReactProp(name = "opacity")
    override fun setOpacity(polyline: AMapPolyline, opacity: Float) {
        polyline.setOpacity(opacity)
    }

    @ReactProp(name = "geodesic")
    fun setGeodesic(polyline: AMapPolyline, geodesic: Boolean) {
        polyline.setGeodesic(geodesic)
    }

    @ReactProp(name = "dottedLine")
    fun setDottedLine(polyline: AMapPolyline, dottedLine: Boolean) {
        polyline.setDottedLine(dottedLine)
    }

    @ReactProp(name = "gradient")
    fun setGradient(polyline: AMapPolyline, gradient: Boolean) {
        polyline.setGradient(gradient)
    }
}
